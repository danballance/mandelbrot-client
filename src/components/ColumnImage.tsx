import * as React from "react";
import { Grid, Dimmer, Loader } from 'semantic-ui-react'

interface Props {
  size: number,
  url: string,
  imageLoaded: boolean
  onImageLoad: (event: React.SyntheticEvent) => void
  onMouseMove: (event: React.MouseEvent) => void
  onImageClicked: (event: React.MouseEvent<HTMLImageElement>) => void
}

export const ColumnImage: React.FC<Props> = (props: Props) => {
  const imageStyles = {
    width: props.size + 'px',
    height: props.size + 'px'
  };
  const loadingStyles = {
    display: (props.imageLoaded) ? "none" : "block"
  };
  const hiddenStyles = {
    display: (props.imageLoaded) ? "block" : "none"
  };
  return (
    <Grid.Column mobile={16} computer={8} textAlign="right">
      <div className="imageWrapper" style={imageStyles}>
        <div className="loading" style={loadingStyles}>
          <Dimmer active>
            <Loader inverted size='massive'>Loading</Loader>
          </Dimmer>
        </div>
        <img src={props.url}  
             onLoad={props.onImageLoad}  
             style={hiddenStyles} 
             onMouseMove={props.onMouseMove} 
             onClick={props.onImageClicked} />
      </div>
    </Grid.Column>
  );
}
