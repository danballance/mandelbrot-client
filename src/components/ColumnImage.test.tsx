import React from 'react';
import { shallow, mount } from 'enzyme'
import { ColumnImage } from './ColumnImage'

const onImageLoad = jest.fn();
const onMouseMove = jest.fn();
const onImageClicked = jest.fn();

const columnImage = mount(<ColumnImage 
    size={400}
    url="https://localhost/api"
    imageLoaded={false}
    onImageLoad={onImageLoad}
    onMouseMove={onMouseMove}
    onImageClicked={onImageClicked}
/>);

describe('ColumnImage', (): void => {
    it('sets the passed URL as the image src', (): void => {
        const image = columnImage.find('GridColumn div.imageWrapper img');
        expect(image.prop('src')).toBe('https://localhost/api');
    })

    it('sets display=none when props.imageLoaded is false', (): void => {
        const image = columnImage.find('GridColumn div.imageWrapper img');
        expect(image.get(0).props.style).toHaveProperty('display', 'none');
    })

    it('sets display=block when props.imageLoaded is true', (): void => {
        columnImage.setProps({imageLoaded: true});
        const image = columnImage.find('GridColumn div.imageWrapper img');
        expect(image.get(0).props.style).toHaveProperty('display', 'block');
    })

    it('triggers onImageClicked when image is clicked', (): void => {
        columnImage
            .find('GridColumn div.imageWrapper img')
            .simulate('click');
        expect(onImageClicked).toHaveBeenCalled();
    })

    it('triggers onImageLoad when the image loads', (): void => {
        columnImage
            .find('GridColumn div.imageWrapper img')
            .simulate('load');
        expect(onImageLoad).toHaveBeenCalled();
    })

    it('renders correctly', (): void => {
        const columnImage = shallow(<ColumnImage 
            size={400}
            url="https://localhost/api"
            imageLoaded={false}
            onImageLoad={onImageLoad}
            onMouseMove={onMouseMove}
            onImageClicked={onImageClicked}
        />);
        expect(columnImage).toMatchSnapshot();
    })
});
