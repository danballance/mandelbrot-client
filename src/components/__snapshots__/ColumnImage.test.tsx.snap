// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnImage renders correctly 1`] = `ShallowWrapper {}`;
